src/server/mod.rs

```diff
@@ -42,14 +42,42 @@ impl SpaceapiServerBuilder {
 
         SpaceapiServerBuilder {
             status,
-            redis_connection_info: Err("redis_connection_info missing".into()),
+            pool: Err("redis_connection_info missing".into()),
             sensor_specs: vec![],
             status_modifiers: vec![],
         }
     }
 
     pub fn redis_connection_info<R: IntoConnectionInfo>(mut self, redis_connection_info: R) -> Self {
-        self.redis_connection_info = redis_connection_info.into_connection_info().map_err(|e| e.into());
+        let redis_connection_info  = redis_connection_info
+            .into_connection_info();
```

Comment:
In the old version we propagated potential errors to `self.redis_connection_info`, which would then be returned to the user when building the server. However, now you unwrap the result on line 74. This would result in a panic. We should handle those errors somehow.

One approach would be to convert the return type of this method to `Result<Self, SpaceapiServerError>`. However, that means that the user has to do error handling multiple times, first when setting the `redis_connection_info` and again when running `build`.

```rust
SpaceapiServerBuilder::new(status)
  .redis_connection_info(...)
  .expect("Could not connect to database")
  .build()
  .expect("Could not build server");
```

It might be better if errors would get collected and returned by `build` (that's why the `redis_pool` field is a `Result`). From the user's point of view:

```rust
SpaceapiServerBuilder::new(status)
  .redis_connection_info(...)
  .build()
  .expect("Could not build server");
```

The question is whether it's a good idea to start the connection attempt in the `redis_connection_info` method, but to return the error later, during `build`. Maybe another approach might be good?

```rust
enum RedisInfo<C: IntoConnectionInfo> {
    None,
    Pool(r2d2::Pool<r2d2_redis::RedisConnectionManager>),
    ConnectionInfo(C),
    Err(SpaceapiServerError),
}

pub fn redis_connection_info<R: IntoConnectionInfo>(mut self, redis_connection_info: R) -> Self {
    match redis_connection_info.into_connection_info() {
        Ok(ci) => self.redis_info = RedisInfo::ConnectionInfo(ci),
        Err(e) => self.redis_info = Redis:Info::Err(e.into()),
    };
    self
}

pub fn redis_pool(mut self, pool: r2d2::Pool<r2d2_redis::RedisConnectionManager>) -> Self {
    self.redis_info = RedisInfo::Pool(pool);
    self
}
```

You'd initialize the `redis_info` field to `RedisInfo::None`. Then you would match on the actual value during `build`, set up the pool if necessary (in case it's `RedisInfo::ConnectionInfo`) and propagate errors that can happen. (You should be able to handle most errors using `?`, since `SpaceapiServerError` has a `From<RedisError>` implementation.)

Reply:
Hm, I also wasn't happy with starting the connection in `redis_connection_info`! This approach is pretty clever, let's go with that!

Reply:
That is indeed quite a nice solution @dbrgn :+1:

Reply:
I thought some more about this (and actually implemented it): @dbrgn does it actually make sense for `RedisInfo` to hold a `IntoConnectionInfo` trait? It would probably be better to just use a `ConnectionInfo`, since we need to call `into_connection_info()` anyways. Does that make sense?

Reply:
>  It would probably be better to just use a ConnectionInfo, since we need to call into_connection_info() anyways. Does that make sense?

Sounds reasonable to me.

Reply:
Ah, yes, totally! Good point.

Reply:
Great, pushed!